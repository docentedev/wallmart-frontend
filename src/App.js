import Router from './Router'
import useCart from './useCart';

function App() {
  const { cartItems, addItem, removeItem, substractItem, get } = useCart()
  return (
    <div>
      <Router
        substractItem={substractItem}
        addItem={addItem}
        cartItems={cartItems}
        removeItem={removeItem}
        get={get}
      />
    </div>
  );
}

export default App;
