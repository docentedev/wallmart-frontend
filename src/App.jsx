import Router from './Router'
import useFetchDiscounts from './hooks/useFetchDiscounts'

function App() {
  const { discounts } = useFetchDiscounts()

  return discounts && (
    <div>
      <Router />
    </div>
  );
}

export default App;
