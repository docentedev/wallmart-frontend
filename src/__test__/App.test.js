import React from 'react'
import TestRenderer from 'react-test-renderer'

import App from '../App'

jest.mock('../hooks/useFetchDiscounts', () => ({
    __esModule: true,
    default: () => ({ discounts: [{}] }),
}))

jest.mock('../Router', () => ({
    __esModule: true,
    default: () => (<div className="router">Router</div>),
}))

describe('App test', () => {
    it('should render App component', () => {
        const testRenderer = TestRenderer.create(<App />)
        const testInstance = testRenderer.root

        expect(testRenderer.toJSON()).toMatchSnapshot()
        expect(testInstance.findByProps({ className: 'router' }).children).toEqual(['Router'])
    })
})
