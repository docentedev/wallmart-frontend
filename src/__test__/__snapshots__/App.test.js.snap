// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App App 1`] = `
<div
  className="router"
>
  Router
</div>
`;
